tests/test_secondary.py

```diff
@@ -701,6 +724,74 @@ def test_40_process_metadata(self):
 
 
 
+  def test_45_process_partial_metadata(self):
+    """
+    Tests uptane.clients.secondary.Secondary::process_partial_metadata()
+
+    Tests PV Secondary client in 2 situations:
+     - secondary_instances[3]: Director's targets metadata available with 
+       valid signatures in JSON mode
+     - secondary_instances[3]: Director's targets metadata available with 
+       invalid signatures in JSON mode
+     - secondary_instances[3]: Director's targets metadata available with 
+       valid signatures in DER mode
+    """
+    # --- Test this test module's setup (defensive)
+    # First, check the source directories, from which the temp dir is copied.
+    # This first part is testing this test module, since this setup was done
+    # above in setUpClass(), to maintain test integrity over time.
+    # We should see only root.(json or der).
+    for data_directory in [
+        TEST_DIRECTOR_METADATA_DIR, TEST_IMAGE_REPO_METADATA_DIR]:
+
+      self.assertEqual(
+          ['root.der', 'root.json'],
+          sorted(os.listdir(data_directory)))
+    
+    sample_working_metadata_path = os.path.join(uptane.WORKING_DIR, 'samples', 
+        'sample_pv_secondary_target_metadata.json')
+
+    sample_bad_sig_working_metadata_path = os.path.join(uptane.WORKING_DIR,
+        'samples', 'sample_pv_secondary_target_metadata_bad_sig.json')
+
+    sample_der_working_metadata_path = os.path.join(uptane.WORKING_DIR, 'samples', 
+        'sample_pv_secondary_target_metadata.der')
+
+    pv_secondary_dir = TEMP_CLIENT_DIRS[PV_SECONDARY1_INDICE]
+    instance = secondary_instances[PV_SECONDARY1_INDICE]
+    director_targets_metadata_path_json = os.path.join(pv_secondary_dir, 
+        'metadata', 'director_targets.json')
+
+    director_targets_metadata_path_der = os.path.join(pv_secondary_dir, 
+        'metadata', 'director_targets.der')
+
+    # PV Secondary 1 with valid director public key and update JSON
+    shutil.copy(sample_working_metadata_path, director_targets_metadata_path_json)
+    if not os.path.exists(director_targets_metadata_path_json):
+      raise("Director's targets not available")
```

Comment:
First, this line doesn't work: you can't raise an exception this way.

Further, test code shouldn't generally be raising its own exceptions. You should perform a test using self.assertXYZ(something). In special circumstances, if something is very wrong with the test code itself (like this), you might want to have normal assert statements to raise assertion errors (debatable) instead of failing the test. Here, though, I probably wouldn't do either of those things: there's no reason to suppose that shutil.copy(X,Y) will fail to copy X to Y without raising an exception.

So I'll just chop these six lines (if: raise; if: raise; if: raise).